src/pint/models/stand_alone_psr_binaries/BT_piecewise.py

```diff
@@ -0,0 +1,441 @@
+
+                if key[0:4] == "T0X_" or key[0:4] == "A1X_":
+                    piece_index.append((key[4:8]))  # appends index to array
+            piece_index = np.unique(
+                piece_index
+            )  # makes sure only one instance of each index is present returns order indeces
+            for (
+                index
+            ) in (
+                piece_index
+            ):  # looping through each index in order they are given (0 -> n)
+                param_pieces = (
+                    []
+                )  # array to store specific piece i's information in the order [index,T0X,A1X,Group's lower edge, Group's upper edge,]
+                piece_number = f"{int(index):04d}"
+                # print(piece_number)
+                param_pieces.append(piece_number)
+                string = [
+                    "T0X_" + index,
+                    "A1X_" + index,
+                    "PLB_" + index,
+                    "PUB_" + index,
+                ]
+
+                # print(f"Should be the last trace :{valDict}")
+
+                if string[0] not in param_pieces:
+                    for i in range(0, len(string)):
+                        # print(string[i])
+                        if string[i] in valDict:
+                            param_pieces.append(valDict[string[i]])
+                        elif string[i] not in valDict:
+                            attr = string[i][0:2]
+                            param_pieces.append(getattr(self, attr))
+                            # Raises error if range not defined as there is no Piece upper/lower bound in the model.
+                # string = ["PLB_"+index,"PUB_"+index]
+                # for i in range(0,len(string)):
+                #    #print(string[i])
+                #    if string[i] in valDict:
+                #        param_pieces.append(valDict[string[i]])
+                #    elif string[i] not in valDict:
+                #        attr = string[i][0:2]
+                #        param_pieces.append(getattr(self, attr))
+
+                # print(param_pieces)
+                piecewise_parameter_information.append(param_pieces)
+            self.valDict = valDict
+            self.piecewise_parameter_information = sorted(
+                piecewise_parameter_information, key=lambda x: x[3]
+            )  # sorts the array chronologically by lower edge of each group,correctly works for unordered pieces (i.e. index 0 can correspond to an arbitrary group of data at any time)
+            # print(f"Filled information array: {self.piecewise_parameter_information}")
+            if len(self.piecewise_parameter_information) > 0:
+                # check = hasattr(self,"t")
+                # print(f"Piecewise parameter loader can see t: {check}")
+                if hasattr(self, "_t") is True:
```

Comment:
Don't use `is True`; just say `if hasattr(...):`

Reply:
Also if you set this to None in `__init__` you can avoid the need tor `hasattr`.

Reply:
removed instances of is True. There may still be some `hasattr`'s in other conditions that could follow the same treatment